website/docs/reference/commands/deps.md

```diff
@@ -62,7 +62,11 @@ Update your versions in packages.yml, then run dbt deps
 
 dbt generates the `package-lock.yml` file in the _project_root_ where `packages.yml` is recorded, which contains all the resolved packages, the first time you run `dbt deps`. Each subsequent run records the packages installed in this file. If the subsequent `dbt deps` runs contain no updated packages in `dependencies.yml` or `packages.yml`, dbt-core installs from `package-lock.yml`. 
 
-When you update the package spec and run `dbt deps` again, the package-lock and package files update accordingly. You can run `dbt deps --lock` to update the `package-lock.yml` with the most recent dependencies from `packages`.
+When you update the package spec and run `dbt deps` again, the `package-lock.yml` and `packages.yml` files update accordingly. 
+
+There are two flags related to `package-lock.yml`:
+- `dbt deps --upgrade` &mdash; creates or updates the `package-lock.yml` file with the most recent dependencies from `packages.yml` and installs them.
+- `dbt deps --lock` &mdash; creates or updates the `package-lock.yml` file but does not install the packages. 
```

Comment:
A key thing to be clear about is that `--upgrade` will not install any packages if `--lock` is also passed.

So I'd suggest flipping the order of the flags and adding a caveat:
```suggestion
- `dbt deps --lock` &mdash; creates or updates the `package-lock.yml` file but does not install the packages.
- `dbt deps --upgrade` &mdash; creates or updates the `package-lock.yml` file with the most recent dependencies from `packages.yml`. Also install the packages unless the `--lock` flag is also passed.
```

### Behaviors with different combination of flags

Put another way, `--upgrade --lock` is equivalent to just `--lock`. Although it doesn't hurt anything to use both at the same time, it makes most sense in practice to use either one or the other (or neither).

When neither is passed, the lock file shouldn't change as long as the package spec hasn't changed (even if a newer version of the package has been released in the meantime).

If the package spec has changed, my understanding is that the lock file will _always_ be updated accordingly no matter which flags are / aren't included.

### Purpose of a lock file

The big idea of having a lock file is:
- repeatable builds (without risk of being affected by an upstream dependency that released a breaking change)

If the lock file changes in source code version control (or during continuous deployment due to a change to `packages.yml` / `dependencies.yml`), a breaking change from an upstream dependency can still affect the project.